plasma_framework/contracts/src/utils/RLPReader.sol

```diff
@@ -135,102 +178,85 @@ library RLPReader {
     }
 
     /**
-     * @notice Decodes the RLPItems length from a bytes array.
+     * @notice Decodes the RLPItem's length and offset.
      * @dev This function is dangerous !!! Ensure that the returned length is within the bounds of where memPtr points to.
      * @param memPtr Pointer to the dynamic bytes array in memory
-     * @return The encoded RLPItem length
+     * @return The length of the RLPItem (including the length field) and the offset of the payload
      */
-    function decodeItemLengthUnsafe(uint256 memPtr) internal pure returns (uint256) {
-        uint256 decodedItemLengthUnsafe;
+    function decodeLengthAndOffset(uint256 memPtr) internal pure returns (uint256, uint256) {
```

Comment:
Can we just pass in the `RLPItem` and also do the length check done in the RLP spec python example?

https://github.com/ethereum/wiki/wiki/RLP

Reply:
I don't think it easy to do it like that. In the python example they use string manipulation with substr and concatenation, but this isn't so easy in Solidity. Memory pointers seem to be a better fit.

Reply:
oh, I am suggesting about the memory pointer part. But if just compare the two functions, the python example also has conditions like `length > prefix - 0x80` in the branchings. If we pass in the RLPItem, I think we can use the `item.len` to match the logic in python.

Reply:
They're different. In python `length = len(input)`, but in Solidity you only have the length of the overall buffer. You don't get the length of each item (until you decode it)

Reply:
I mean, in python `length` is the length of the remainder of the buffer, which is not the same as the `item.len`.

Reply:
okay, let me try to understand the assemblies again 😅